astropy/units/format/base.py

```diff
@@ -51,11 +63,11 @@ def format_exponential_notation(cls, val, format_spec="g"):
         return format(val, format_spec)
 
     @classmethod
-    def _format_superscript(cls, number):
+    def _format_superscript(cls, number: str) -> str:
         return f"({number})" if "/" in number or "." in number else number
 
     @classmethod
-    def _format_unit_power(cls, unit, power=1):
+    def _format_unit_power(cls, unit: UnitBase, power: Real = 1) -> str:
```

Comment:
I assume the annotation for `unit` here needs to stay consistent with that in `_get_unit_name`, just in case my first comment leads to a change there.